File: 2 семестр/лаб5/z2.py
'''Описати функцію Calc (A, B, Op) дійсного типу, що виконує над ненульовими
дійсними числами A та B одну з арифметичних операцій і повертає її результат. Вид
операції визначається цілим параметром Op: 1 - віднімання, 2 - множення, 3 - ділення,
інші значення - додавання. За допомогою Calc виконати для даних A і B операції, які
визначаються даними цілими N1, N2, N3.'''

def Calc (a, b, op):
    if op==1:
        return a-b
    elif op==2:
        return a*b
    elif op==3:
        return a/b
    else:
        return a+b

print('enter A, B and operation number (1-, 2*, 3/, other+ )')
a=int(input())
b=int(input())
op=int(input())

print('result: ',Calc (a, b, op))

File: 1 семестр/Лабораторна 9/z-2.cpp
#include <iostream>
using namespace std;
int main()
{
    int** dArray, length_of_array;

    cout<< " enter length of array a[n][n]  n: ";
    cin >> length_of_array;

    dArray = new int* [length_of_array];

    for (int i = 0; i < length_of_array; i++)
        dArray[i] = new int[length_of_array];


    cout << " enter array "<<endl;
    for (int i = 0; i < length_of_array; i++)
        for (int j = 0; j < length_of_array; j++)
            cin >> dArray[i][j];
   
 int k, p,sum1=0,sum2=0;
    cout <<endl<< "      Find the sum of the elements " <<endl<<endl<<
        " enter number k - th column : ";
    cin >> k;
    cout<<" and p - th row : ";
    cin >> p;
    
    for (int i = 1; i <= length_of_array; i++)
    {
        for (int j = 1; j <= length_of_array; j++)
        {
            if (k == j )
                sum1=sum1+ dArray[i-1][j-1];
            if (p == i)
                sum2 = sum2 + dArray[i-1][j-1];
        }
    }

    cout<<endl << "      RESULT "<<endl;
    for (int i = 0; i < length_of_array; i++)
    {
        for (int j = 0; j < length_of_array; j++)
        {
            if (j >= i)
                dArray[i][j] = 0;
            cout << dArray[i][j] << '\t';
        }
        cout << endl;
    }
    cout << " sum "<<k<<" column = " << sum1<<endl;
    cout << " sum "<<p<<" row = " << sum2 << endl;

    system("pause");
    return 0;
}
